aws/src/integration/java/org/apache/iceberg/aws/glue/DynamoLockManagerTest.java

```diff
@@ -0,0 +1,199 @@
+        .collect(Collectors.toList());
+    Assert.assertEquals(1, results.stream().filter(s -> s).count());
+  }
+
+  @Test
+  public void testWaitLock_singleProcess() throws Exception {
+    boolean succeed = lockManager.tryLock(tableId(tableName),
+        AwsProperties.LOCK_EXPIRE_MS_DEFAULT,
+        AwsProperties.LOCK_WAIT_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put("lockId", AttributeValue.builder().s(tableId(tableName)).build());
+    try {
+      dynamo.getItem(GetItemRequest.builder()
+          .tableName(lockTableName)
+          .key(key)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when getting lock after insert");
+    }
+  }
+
+  @Test
+  public void testWaitLock_multiProcess_allSucceed() {
+    List<Boolean> results = IntStream.range(0, 100).parallel()
+        .mapToObj(i -> {
+          try {
+            return lockManager.tryLock(tableId(tableName), 10, 100000);
+          } catch (InterruptedException e) {
+            LOG.error("lock acquire interrupted", e);
+            return false;
+          }
+        })
+        .collect(Collectors.toList());
+    Assert.assertEquals(100, results.stream().filter(s -> s).count());
+  }
+
+  @Test
+  public void testWaitLock_multiProcess_allFail() {
+    List<Boolean> results = IntStream.range(0, 100).parallel()
+        .mapToObj(i -> {
+          try {
+            return lockManager.tryLock(tableId(tableName), 1000000, 10);
+          } catch (InterruptedException e) {
+            LOG.error("lock acquire interrupted", e);
+            return false;
+          }
+        })
+        .collect(Collectors.toList());
+    Assert.assertEquals(1, results.stream().filter(s -> s).count());
+  }
+
+
+  @Test
+  public void testUnlock() {
+    boolean succeed = lockManager.tryLock(tableId(tableName),
+        AwsProperties.LOCK_EXPIRE_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put("lockId", AttributeValue.builder().s(tableId(tableName)).build());
+    dynamo.deleteItem(DeleteItemRequest.builder()
+        .tableName(lockTableName)
+        .key(key)
+        .build());
+    succeed = lockManager.tryLock(tableId(tableName),
+        AwsProperties.LOCK_EXPIRE_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+  }
+
+  @Test
+  public void testTableCreationFailure() {
+    DynamoDbClient dynamo2 = Mockito.mock(DynamoDbClient.class);
+    Mockito.doThrow(ResourceNotFoundException.class).when(dynamo2)
+        .describeTable(Mockito.any(DescribeTableRequest.class));
+    AssertHelpers.assertThrows("should fail to initialize the lock manager",
+        IllegalStateException.class,
+        "Cannot find Dynamo table",
+        () -> new DynamoLockManager(dynamo2, lockTableName));
+  }
+
+  private static String tableId(String tableName) {
```

Comment:
Considering the lock only uses an ID string, I don't think that this method is needed. You could just use a static ID string.